### PR TITLE
[NetManager] Migrate to Site Registry 

### PR DESCRIPTION
**Pull request by ja-odur**
#### Summary of Changes (What does this PR do?)
- Updates device table to include site information
- Changes `locationRegistry` to `siteRegistry`
- Adds a site detail page (click on any site on the table)
- Update device deployment to include `site_id`
- Updates the device recall url
- Fixes this [issue](https://github.com/airqo-platform/AirQo-frontend/issues/263)

#### Is this change ready to hit production in its current state?
- [x] Yes
- [ ] No

#### Status of maturity (all need to be checked before merging):

- [x] I consider this code done
- [ ] I've tested this locally
- [ ] This branch has been LGTM'ed by a reviewer

#### How should this be manually tested?
* Fetch and checkout to this branch locally
* Install requirements `npm install`
* Start netmanager application `npm run stage-mac` (macOS) or `npm run stage-pc` (windows Platform)
* Confirm the summary of changes listed above

#### What are the relevant tickets?
- [PLAT-642](https://airqoteam.atlassian.net/browse/PLAT-642)

#### Screenshots (optional)

### CODE REVIEW

File: netmanager/src/views/components/DataDisplay/DeviceView/DeviceDeployStatus.js [hunk 1]
```diff
@@ -332,27 +355,49 @@ export default function DeviceDeployStatus({ deviceData }) {
     }
   };
 
+  const createSiteOptions = () => {
+    const options = [];
+    sites.map((site) =>
+      options.push({
+        value: site._id,
+        label: site.name || site.description || site.generated_name,
```
Comment (Baalmart) [inline netmanager/src/views/components/DataDisplay/DeviceView/DeviceDeployStatus.js:363]: lovely!

**Review by Baalmart — CHANGES_REQUESTED** [review]
This is beyond lovely! @ja-odur 
For a good user experience, please add a back navigation on the site registry page accordingly. From user (Deo Okure) experience on device registry page, this is actually much needed so I will go ahead and request this change. :)

**Comment by danielogen** [comment]
Everything looks great @ja-odur. Let me wait for the change @Baalmart requested and approve :)

File: netmanager/src/views/components/DataDisplay/DeviceView/DeviceDeployStatus.js (hunk 1 above)
Comment (ja-odur) [reply netmanager/src/views/components/DataDisplay/DeviceView/DeviceDeployStatus.js:363]: Thia

> This is beyond lovely! @ja-odur
> For a good user experience, please add a back navigation on the site registry page accordingly. From user (Deo Okure) experience on device registry page, this is actually much needed so I will go ahead and request this change. :)
> 
> See screenshot
> Screenshot 2021-06-24 at 20 40 44

This has been resolved @Baalmart

**Review by Baalmart — APPROVED** [review]
LGTM

**Review by NoahNsimbe — APPROVED** [review]
LGTM

**Pull request merged**